server/src/test/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoRequestTests.java

```diff
@@ -0,0 +1,132 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.cluster.node.info;
+
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Granular tests for the {@link NodesInfoRequest} class. Higher-level tests
+ * can be found in {@link org.elasticsearch.rest.action.admin.cluster.RestNodesInfoActionTests}.
+ */
+public class NodesInfoRequestTests extends ESTestCase {
+
+    /**
+     * Make sure that we can set, serialize, and deserialize arbitrary sets
+     * of metrics.
+     */
+    public void testMetricsSetters() throws Exception {
+        NodesInfoRequest request = new NodesInfoRequest(randomAlphaOfLength(8));
+        request.settings(randomBoolean());
+        request.os(randomBoolean());
+        request.process(randomBoolean());
+        request.jvm(randomBoolean());
+        request.threadPool(randomBoolean());
+        request.transport(randomBoolean());
+        request.http(randomBoolean());
+        request.plugins(randomBoolean());
+        request.ingest(randomBoolean());
+        request.indices(randomBoolean());
+        NodesInfoRequest deserializedRequest = roundTripRequest(request);
+        assertThat(request.settings(), equalTo(deserializedRequest.settings()));
+    }
+
+    /**
+     * Test that a newly constructed NodesInfoRequestObject requests all of the
+     * possible metrics defined in {@link NodesInfoRequest.Metrics}.
+     */
+    public void testNodesInfoRequestDefaults() {
+        NodesInfoRequest defaultNodesInfoRequest = new NodesInfoRequest(randomAlphaOfLength(8));
+        NodesInfoRequest allMetricsNodesInfoRequest = new NodesInfoRequest(randomAlphaOfLength(8));
+        allMetricsNodesInfoRequest.all();
+
+        assertRequestsEqual(defaultNodesInfoRequest, allMetricsNodesInfoRequest);
+    }
+
+    /**
+     * Test that the {@link NodesInfoRequest#all()} method sets all of the
+     * metrics to {@code true}.
+     */
+    public void testNodesInfoRequestAll() throws Exception {
+        NodesInfoRequest request = new NodesInfoRequest("node");
+        request.all();
+
+        assertTrue(request.settings());
```

Comment:
You could loop through `Metric.values()` instead of hardcoding each to check for?

Reply:
I'll add some TODOs to this effect, since I will definitely be doing this in a follow-up.